File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        i = 0
        product = 1
        while i < len(nums):
            for index, num in enumerate(nums):
                if i == index:
                    continue
                else:
                    product *= num 
            res.append(product)
            product = 1
            i+=1
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-0.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var seen = [Int: Int]()

        for (index, num) in nums.enumerated() {
            let needed = target - num

            if let previousIndex = seen[needed] {
                return [previousIndex, index]
            }

            seen[num] = index
        }

        return []
    }
}